### DISCUSSION TITLE
Automatically building an index page

### DISCUSSION THREAD

**Discussion by pitosalas** [mkdocs/mkdocs#3535] [Q&A]
Is this the right place to ask a question? If not, I apologize, please direct me. 

I am just starting with Mkdocs. I am converting my Wordpress blog into a static site using MkDocs.

Is there a feature which will organize a directory with several hundred posts (.md) files and automatically generate index pages or something along those lines. Or do I generate that myself?

**Comment by oprypin** [comment]
* Might be the same as https://github.com/mkdocs/mkdocs/discussions/3077

**Comment by oprypin** [comment]
Perhaps through extra_templates

https://www.mkdocs.org/user-guide/configuration/#extra_templates

you could create a page with this content

https://github.com/waylan/mkdocs-nature/blob/master/nature/sitemap.html

**Comment by oprypin** [comment]
This actually gives me an idea- I'll make a plugin to do this

**Comment by danielo515** [comment]
Any easy way of doing this? I just want to have an index when I don't have one (basically, i want an index in every folder containing files)

**Reply by santi-mir** [reply]
I am unsure what OP asks. What I did was to write a little bash script that builds the index:

```bash
find <pyfiles_folder> -type f -name *.py -not -path "**/tests/**" -not -name "__init__.py" | sed --expression="s/\//./g" --expression="s/^/::: /g" --expression="s/.py$//g" > docs/in
dex.md
```

It ignores test files and writes an index file with the `::: path.to.module` also ignores `__init__` since it's either automatically searched or ignored. Doesn't work explicitly.

**Reply by danielo515** [reply]
Turns out that the theme I'm using (material for mkdocs) has a functionality to build a "blog", and that automatically creates the required pages. In my case, this fit my needs

**Reply by santi-mir** [reply]
ill check it out, thanks!
